generators/rest-adapter-generator/src/test/java/uk/gov/justice/config/GeneratorPropertiesFactory.java

```diff
@@ -0,0 +1,25 @@
+package uk.gov.justice.config;
+
+import static uk.gov.justice.services.test.utils.core.reflection.ReflectionUtil.setField;
+
+import uk.gov.justice.maven.generator.io.files.parser.core.GeneratorProperties;
+import uk.gov.justice.services.generators.commons.config.CommonGeneratorProperties;
+
+public class GeneratorPropertiesFactory {
+
+    private static final String SERVICE_COMPONENT_KEY = "serviceComponent";
+
+    public static GeneratorPropertiesFactory generatorProperties() {
+        return new GeneratorPropertiesFactory();
+    }
+
+    public GeneratorProperties withServiceComponentOf(final String serviceComponent) {
+        final CommonGeneratorProperties properties = new CommonGeneratorProperties();
+        try {
+            setField(properties, SERVICE_COMPONENT_KEY, serviceComponent);
+        } catch (IllegalAccessException e) {
+            throw new RuntimeException(e);
```

Comment:
If this is just a test class for one test no problem, but if this class gets promoted and reused, then this exception needs a message. Something like 'Failed to set field 'propertyName' on class 'className'